whisper.cpp

```diff
@@ -2539,6 +2574,13 @@ static struct ggml_cgraph * whisper_build_graph_decoder(
 
     struct ggml_tensor * logits = ggml_mul_mat(ctx0, model.d_te, cur);
 
+    // FIXME: maybe this additional node should by part of pipeline if requested,
+    // since it is only needed for dtw timestamps and useless on all other decoding
+    // calls. Should be counted when allocating memory for the decoder graph though.
+    // [EXPERIMENTAL] Token-level timestamps with DTW
+    for (int il = 0; il < n_layer; ++il)
+        ggml_build_forward_expand(gf, wstate.cross_QKs[il]);
```

Comment:
We can do this https://github.com/ggerganov/ggml/issues/700#issuecomment-1897621084

> If you want to keep some of the intermediate results, the recommended approach would be pre-allocate some tensors in a different buffer and use ggml_cpy to copy the result there.

Reply:
Not sure if this changes anything for this case, but I plan to add (relatively soon) a way to specify graph outputs by setting a flag in `ggml_tensor`. Setting the "output" flag would prevent it from being overwritten by future operations when using ggml-alloc.